core/src/test/java/com/backblaze/b2/json/B2JsonTest.java

```diff
@@ -2792,6 +2793,104 @@ public void testClassWithLotsOfFields() throws IOException, B2JsonException {
         assertEquals("b2", obj3.field77);
     }
 
+    /**
+     * Cannot directly test the case with MAX_ARRAY_SIZE being (Integer.MAX_VALUE - 8)
+     * since the real OutOfMemory error would occur without triggering the intended
+     * IOException ("Requested array size exceeds maximum limit"). This happens even
+     * when heap size is increased to 8GB. This happens probably because the JVM heap
+     * is probably heavily allocated before the OutputStream array size is anywhere
+     * close to half of MAX_ARRAY_SIZE.
+     *
+     * Instead, create a subclass of B2JsonByteArrayOutputStream with lowered
+     * MAX_ARRAY_SIZE (1000) for testing purpose: IOException will then be thrown
+     */
+    private static class B2JsonByteArrayOutputStreamForTest extends B2JsonByteArrayOutputStream {
+        private static final int MAX_ARRAY_SIZE = 1000;
+
+        @Override
+        protected void grow(int minCapacity) throws IOException {
```

Comment:
i like that you've got a way of testing the boundary condition in a unit test.

could we do it without having to make a copy of grow() which may get out-of-sync with the real implementation?

maybe we should make "maxCapacity" be a member and have it default to Integer.MAX_VALUE.  And this subclass would just pass a non-default maxCapacity?  yes, it would take a little bit of extra space, but it's not like we have millions of these at a time and it _does_ make it more testable.

whatcha think? got a better proposal?  :)